### PR TITLE
feat: append links to frontmatter properties

### LINKED ISSUE #768: [BUG] Append link to current file in properties
**Describe the bug**
I use the Append link option to add the link to the created file in the current file.
image

It works properly when the cursor is in the body of the current file but when the cursor is in a property field, the link is not append to the property field but on the first line of the body (just after the YAML section).

image

**Expected behavior**
Append the link at the cursor position, even if the cursor is in a property field.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows, Mac
 - Browser Edge, Safari

### PR DESCRIPTION
**Pull request by chhoumann**
## Summary

This consolidates the work from #1276 and #1340 into one maintained PR for #768.

The core feature here is Arjan's configurable append-link placement from #1276: Template and Capture choices can now insert the created/captured-file link into a named frontmatter property on the active note. Arjan's implementation was the right product direction for this issue; I ported that work onto current `master`, adapted it to the shared `AppendLinkSetting.svelte` settings UI, and kept the focused-property caret fix from #1340/#1394 as the fallback for normal append-link placements.

Credit where it is due:
- @ArjanSeijs did the substantial feature work for the explicit **In frontmatter property** placement, including the settings direction and the frontmatter value-shape handling that this PR now carries forward.
- @Ivikhostrup did the live Obsidian investigation for #768 and established the important `processFrontMatter` direction for the focused-property caret bug.

This PR supersedes #1276 and #1340. I am not pushing to either contributor branch.

## Problem

The underlying problem is broader than the issue title. Obsidian Properties are separate editable controls from the Markdown editor, so QuickAdd cannot assume that the Markdown editor cursor is the user's current insertion target. Users need two coherent behaviors:

1. A deliberate setting for writing append links into a chosen frontmatter property.
2. A fallback for the #768 caret case where a normal append-link placement starts while focus is inside an editable Obsidian Properties value.

## Design

- Add `Link placement -> In frontmatter property` for Template and Capture append links.
- Add a required `Frontmatter property` setting and three handling modes, with `Create or convert` as the default:
  - `Create or convert`: create the property when absent, or convert existing scalars to a list before appending.
  - `Create if missing`: create the property when absent, but reject existing scalar/object values.
  - `Require list`: append only to an existing list, treating empty/null as an empty list.
- Persist frontmatter changes through `app.fileManager.processFrontMatter` instead of mutating Obsidian's property DOM.
- Preserve property key casing and reject ambiguous keys that normalize to the same name.
- Keep embeds unsupported for frontmatter placement; YAML receives a normal wikilink string.
- Compose with the `master` append-link destination feature by keeping `Specified note` destinations as bottom-appends to that specified note; frontmatter placement is intentionally scoped to `Current note` because Obsidian Properties focus only identifies the active note.
- Preserve frontmatter settings when switching append-link modes, including disable/re-enable.
- Keep the focused-property #768 fallback only for non-`In frontmatter property` placements. Explicit configuration wins over the currently focused property.
- Carry focused-property context through QuickAdd suggesters, including top-level `Run QuickAdd` and Multi-choice child selection, so opening the modal does not erase the Properties-field target.
- Keep append-link insertion post-commit, matching existing QuickAdd behavior. If frontmatter append fails after a Template/Capture has already created or written its target file, QuickAdd logs/reports that error but preserves the successful primary side effect for URI/automation callers so retries do not duplicate notes or captures.

## Validation

Automated gates:
- `pnpm run build-with-lint` passes.
- `pnpm run check` passes with 0 errors and the existing 4 warnings.
- `pnpm run test` passes: 237 files passed, 5 skipped; 2875 tests passed, 37 skipped.
- `git diff --check` passes.

Targeted regression coverage added for:
- `LinkPlacement` normalization and frontmatter support detection.
- Frontmatter helper behavior for arrays, missing/null/empty properties, scalars, objects, casing, and ambiguous keys.
- `insertFileLinkToActiveView` writing through `processFrontMatter` and propagating insertion failures to the engine.
- Append-link settings defaulting frontmatter handling to `Create or convert`, preserving `frontmatterProperty`/`frontmatterHandling` across mode changes, including disable/re-enable.
- Template/Capture engine post-commit behavior when configured frontmatter insertion fails.
- Canvas-triggered Capture does not silently skip explicit frontmatter placement.
- Focused Properties-field context surviving top-level and nested/Multi QuickAdd suggesters.

Live Obsidian validation in this worktree's isolated vault:
- Template with `In frontmatter property` + `Create if missing` wrote the created-file link into `related` on the active note.
- Capture with `In frontmatter property` + `Create if missing` wrote the captured-file link into `captures` on the active note and wrote the capture body.
- Default `Create or convert` handling converted a scalar `related: "[[Existing]]"` into a list and appended the new link.
- `Require list` treated `related: ""` as an empty list and appended the new link.
- Explicit configured property won over a simultaneously focused different property.
- Normal `replaceSelection` append-link still used the focused-property fallback for the original #768 case.
- A command-backed Multi choice with a child Template, selected through the real suggester UI while the `focused` Properties field was active, created the child note and updated the host note to `focused: seed [[Created child ...]]`.
- The top-level `quickadd:runQuickAdd` suggester path, opened while the `focused` Properties field was active, created the selected Template note and updated the host note to `focused: seed [[Created root ...]]`.
- `Require list` with a missing property left the note unchanged and reported the expected QuickAdd error.
- After rebasing onto latest `master` (`19b52f83`), repeated the focused-property live smoke: with focus inside the `related` Properties value, `quickadd:run` created the Template note, appended `[[Created_...]]` into `Related`, and left the note body unchanged.
- After clearing the expected failure and final smoke seeds, `pnpm run obsidian:e2e -- dev:errors` reported no captured runtime errors.

## Migration / Release Notes

No migration is required. Existing boolean append-link settings still normalize at runtime. The new frontmatter fields are optional and only used when placement is `In frontmatter property`.

Fixes #768
Supersedes #1276
Supersedes #1340

## Summary by CodeRabbit

* **New Features**
  * Added **In frontmatter property** placement for Capture and Template, including configurable handling when the target frontmatter is missing or not a list (**error**, **createProperty**, **alwaysAppend**).
  * Added **newLine** placement support and new UI controls for frontmatter property and handling.
* **Bug Fixes**
  * Improved link insertion when a frontmatter Properties field or canvas context has focus, avoiding stale cursor behavior.
  * Ensured frontmatter/link insertion completes before follow-up actions.
* **Documentation**
  * Updated Capture/Template docs with placement rules and focus behavior.
* **Tests**
  * Expanded automated coverage for frontmatter modes and focus/cursor edge cases.

### CODE REVIEW

File: src/choiceExecutor.ts [hunk 1]
```diff
@@ -90,6 +111,7 @@ export class ChoiceExecutor implements IChoiceExecutor {
 			throw error;
 		} finally {
 			this.pendingResult = savedResult;
+			this.endExecutionContext();
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/choiceExecutor.ts:118]: Addressed in 67f8b16b. Focused Properties-field context is now captured before the suggester opens, passed through top-level/nested/Multi ChoiceSuggester options, and applied through ChoiceExecutor.executeWithFocusedProperty when the selected child runs. Added regression coverage for both explicit captured context and top-level default capture, plus live Obsidian validation for a command-backed Multi child and the top-level runQuickAdd path. This SHA replaced 05497a26 in an attribution-only history rewrite so the original contributors receive commit credit.
